src/sdk/pynni/nni/compression/torch/pruning/amc/amc_pruner.py

```diff
@@ -0,0 +1,325 @@
+
+            # [optional] save intermideate model
+            if num_episode / 3 <= 1 or episode % int(num_episode / 3) == 0:
+                agent.save_model(output)
+
+            # update
+            step += 1
+            episode_steps += 1
+            episode_reward += reward
+            observation = deepcopy(observation2)
+
+            if done:  # end of episode
+                print(
+                    '#{}: episode_reward:{:.4f} acc: {:.4f}, ratio: {:.4f}'.format(
+                        episode, episode_reward,
+                        info['accuracy'],
+                        info['compress_ratio']
+                    )
+                )
+                self.text_writer.write(
+                    '#{}: episode_reward:{:.4f} acc: {:.4f}, ratio: {:.4f}\n'.format(
+                        episode, episode_reward,
+                        info['accuracy'],
+                        info['compress_ratio']
+                    )
+                )
+                final_reward = T[-1][0]
+                # print('final_reward: {}'.format(final_reward))
+                # agent observe and update policy
+                for _, s_t, s_t1, a_t, done in T:
+                    agent.observe(final_reward, s_t, s_t1, a_t, done)
+                    if episode > self.ddpg_args.warmup:
+                        agent.update_policy()
+
+                #agent.memory.append(
+                #    observation,
+                #    agent.select_action(observation, episode=episode),
+                #    0., False
+                #)
+
+                # reset
+                observation = None
+                episode_steps = 0
+                episode_reward = 0.
+                episode += 1
+                T = []
+
+                self.tfwriter.add_scalar('reward/last', final_reward, episode)
+                self.tfwriter.add_scalar('reward/best', env.best_reward, episode)
+                self.tfwriter.add_scalar('info/accuracy', info['accuracy'], episode)
+                self.tfwriter.add_scalar('info/compress_ratio', info['compress_ratio'], episode)
+                self.tfwriter.add_text('info/best_policy', str(env.best_strategy), episode)
+                # record the preserve rate for each layer
+                for i, preserve_rate in enumerate(env.strategy):
+                    self.tfwriter.add_scalar('preserve_rate/{}'.format(i), preserve_rate, episode)
+
+                self.text_writer.write('best reward: {}\n'.format(env.best_reward))
+                self.text_writer.write('best policy: {}\n'.format(env.best_strategy))
+        self.text_writer.close()
+
+    def export_pruned_model(self):
+        if self.export_source_path is None:
+            wrapper_model_ckpt = os.path.join(self.output, 'best_wrapped_model.pth')
+        else:
+            wrapper_model_ckpt = self.export_source_path
+        self.env.reset()
+        self.bound_model.load_state_dict(torch.load(wrapper_model_ckpt))
```

Comment:
why exporting the pruned model should load checkpoint?

Reply:
During training, models with best acc reward are saved.  Here we export best searched model.